### PR TITLE
Displacement output background removal + High precision numeric inputs

### PR DESCRIPTION
**Pull request by ricklove**
image

### CODE REVIEW

File: src/controls/widgets/number/InputNumberUI.tsx [hunk 1]
```diff
@@ -34,6 +34,7 @@ export const InputNumberUI = observer(function InputNumberUI_(p: {
     const val = p.value ?? clamp(1, p.min ?? -Infinity, p.max ?? Infinity)
     const mode = p.mode
     const step = p.step ?? (mode === 'int' ? 1 : 0.1)
+    const decimalPoints = Math.ceil(-Math.log10(step))
```
Comment (ricklove) [inline src/controls/widgets/number/InputNumberUI.tsx:37]: Rounding should be smaller than step

File: src/controls/widgets/number/InputNumberUI.tsx [hunk 2]
```diff
@@ -49,32 +50,34 @@ export const InputNumberUI = observer(function InputNumberUI_(p: {
...
-    const syncValues = (value: number | string, soft: boolean = false) => {
+    const syncValues = (value: number | string, roundingModifier = 1, soft: boolean = false) => {
         let num =
             typeof value === 'string' //
                 ? mode == 'int'
                     ? parseInt(value, 10)
                     : parseFloat(value)
                 : value
 
+        const decimalPointsPrecise = Math.ceil(-Math.log10(step * roundingModifier))
```
Comment (ricklove) [inline src/controls/widgets/number/InputNumberUI.tsx:61]: Sub step precision with roundingModifier

File: src/controls/widgets/number/InputNumberUI.tsx [hunk 3]
```diff
@@ -103,7 +106,7 @@ export const InputNumberUI = observer(function InputNumberUI_(p: {
             num = Math.round(num / inverval) * inverval
         }
 
-        syncValues(num, true)
+        syncValues(num, e.shiftKey ? 0.01 : 1, true)
```
Comment (ricklove) [inline src/controls/widgets/number/InputNumberUI.tsx:109]: Hold shift key to get 100x precision with mouse

File: src/controls/widgets/number/InputNumberUI.tsx [hunk 4]
```diff
@@ -273,7 +276,7 @@ export const InputNumberUI = observer(function InputNumberUI_(p: {
                                         return
                                     }
 
-                                    syncValues(ev.currentTarget.value)
+                                    syncValues(ev.currentTarget.value, 0.001)
```
Comment (ricklove) [inline src/controls/widgets/number/InputNumberUI.tsx:279]: Manual entry supports 1000x precision

**Comment by rvion** [comment]
thanks ! 
displacement part is 👌 ✨ 
Regarding input, probably the same, but not 100% sure I fully understand the whole extent of the change.

Anyway, merging right now into dev; we can always revisit / improve the part about inputs if need be (ping @birdddev)

**Pull request merged**